### PR TITLE
Fork and customize the SFTP server image

### PR DESCRIPTION
**Pull request by eyedvabny**
@sna-vien and I figured out how to set the SFTP server to log in JSON, but the change requires changing several files. Rather than juggling that in terraform I've forked the upstream project and made the changes there.

This fork copies over our existing modifications (adding back a key exchange cypher).

### CODE REVIEW

**Comment by eyedvabny** [comment]
Example of JSON logs:
```
{"timestamp":"2023-08-21T22:51:17.995173+00:00", "host":"1e9a0296fd85", "severity":"INFO", "source":"sshd", "pid":"1", "message":" Server listening on :: port 22."}
{"timestamp":"2023-08-21T22:52:41.252979+00:00", "host":"1e9a0296fd85", "severity":"INFO", "source":"sshd", "pid":"40", "message":" Failed password for foo from 172.17.0.1 port 54686 ssh2"}
{"timestamp":"2023-08-21T22:52:48.495872+00:00", "host":"1e9a0296fd85", "severity":"INFO", "source":"sshd", "pid":"40", "message":" Accepted password for foo from 172.17.0.1 port 54686 ssh2"}
{"timestamp":"2023-08-21T22:52:49.048605+00:00", "host":"1e9a0296fd85", "severity":"INFO", "source":"internal-sftp", "pid":"43", "message":" session opened for local user foo from [172.17.0.1]"}
{"timestamp":"2023-08-21T22:52:49.742573+00:00", "host":"1e9a0296fd85", "severity":"INFO", "source":"internal-sftp", "pid":"43", "message":" opendir \"\/\""}
{"timestamp":"2023-08-21T22:52:49.751250+00:00", "host":"1e9a0296fd85", "severity":"INFO", "source":"internal-sftp", "pid":"43", "message":" closedir \"\/\""}
{"timestamp":"2023-08-21T22:52:52.170812+00:00", "host":"1e9a0296fd85", "severity":"INFO", "source":"internal-sftp", "pid":"43", "message":" opendir \"\/upload\""}
{"timestamp":"2023-08-21T22:52:52.180205+00:00", "host":"1e9a0296fd85", "severity":"INFO", "source":"internal-sftp", "pid":"43", "message":" closedir \"\/upload\""}
{"timestamp":"2023-08-21T22:52:53.187924+00:00", "host":"1e9a0296fd85", "severity":"INFO", "source":"internal-sftp", "pid":"43", "message":" session closed for local user foo from [172.17.0.1]"}
{"timestamp":"2023-08-21T22:52:53.203001+00:00", "host":"1e9a0296fd85", "severity":"INFO", "source":"sshd", "pid":"42", "message":" Received disconnect from 172.17.0.1 port 54686:11: disconnected by user"}
{"timestamp":"2023-08-21T22:52:53.203059+00:00", "host":"1e9a0296fd85", "severity":"INFO", "source":"sshd", "pid":"42", "message":" Disconnected from user foo 172.17.0.1 port 54686"}
{"timestamp":"2023-08-21T22:53:12.300410+00:00", "host":"1e9a0296fd85", "severity":"INFO", "source":"sshd", "pid":"44", "message":" Accepted password for foo from 172.17.0.1 port 55782 ssh2"}
{"timestamp":"2023-08-21T22:53:12.845416+00:00", "host":"1e9a0296fd85", "severity":"INFO", "source":"internal-sftp", "pid":"47", "message":" session opened for local user foo from [172.17.0.1]"}
{"timestamp":"2023-08-21T22:53:15.121435+00:00", "host":"1e9a0296fd85", "severity":"INFO", "source":"internal-sftp", "pid":"47", "message":" sent status No such file"}
{"timestamp":"2023-08-21T22:53:18.028575+00:00", "host":"1e9a0296fd85", "severity":"INFO", "source":"internal-sftp", "pid":"47", "message":" sent status No such file"}
```

File: files/rsyslog.conf [hunk 1]
```diff
@@ -20,18 +13,20 @@ $FileCreateMode 0640
 $DirCreateMode 0755
 $Umask 0022
 
+# Log in JSON
```
[the remark this answers is not in the document]
Comment (eyedvabny) [reply files/rsyslog.conf:16]: Yes, you'll need to adapt the logging pipeline. Can't deploy to a cluster until there is an image, which this PR creates

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -28,45 +24,32 @@ jobs:
...
-        if: github.ref == 'refs/heads/master'
-        uses: docker/login-action@v1 
+        if: github.ref == 'refs/heads/main'
+        uses: docker/login-action@v2
         with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```
[the remark this answers is not in the document]
Comment (eyedvabny) [reply .github/workflows/build.yml:49]: it's a default token. There's a GITHUB_TOKEN [injected](https://docs.github.com/en/actions/security-guides/automatic-token-authentication) into every GHA run.

I copied these steps from https://docs.github.com/en/actions/publishing-packages/publishing-docker-images

**Comment by lkanerv** [comment]
IMO forking from Corilus is an anti-pattern that would cause Afresh to incur the maintenance and update costs of the entire server and image hosting, rather than just our config.

We should be able to rely on the image directly from Corilus as-is. That way, we would only version-control and introduce our own config during the bootstrapping process of the SFTP server with Terraform. 

I've archived the repo for now and will leave feedback on that PR. Moving forward, I hope that before such a drastic architectural change like this is PRed, we'd at least have a team convo about it first at something like a tech sync!

**Pull request closed without merging**